modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/mapper/JsonMapper.kt

```diff
@@ -0,0 +1,59 @@
+package com.sphereon.oid.fed.common.mapper
+
+import com.sphereon.oid.fed.openapi.models.EntityStatement
+import com.sphereon.oid.fed.openapi.models.JWTHeader
+import com.sphereon.oid.fed.openapi.models.JWTSignature
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.decodeFromJsonElement
+import kotlin.io.encoding.Base64
+import kotlin.io.encoding.ExperimentalEncodingApi
+
+class JsonMapper {
+
+    /*
+     * Used for mapping JWT token to EntityStatement object
+     */
+    fun mapEntityStatement(jwtToken: String): EntityStatement? {
+        val data = decodeJWTComponents(jwtToken)
+        return if (data.second != null) {
+            Json.decodeFromJsonElement(data.second!!)
+        } else {
+            null
+        }
+    }
+
+    /*
+     * Used for mapping trust chain
+     */
+    fun mapTrustChain(jwtTokenList: List<String>): List<EntityStatement?> {
+        val list: MutableList<EntityStatement?> = mutableListOf()
+        jwtTokenList.forEach { jwtToken ->
+            list.add(mapEntityStatement(jwtToken))
+        }
+        return list
+    }
+
+    /*
+     * Used for decoding JWT to a triple with Header, Payload and Signature
+     */
+    @OptIn(ExperimentalEncodingApi::class)
+    fun decodeJWTComponents(jwtToken: String): Triple<JWTHeader?, JsonElement?, JWTSignature?> {
+        val parts = jwtToken.split(".")
+        if (parts.size != 3) {
```

Comment:
Do we really need to ignore this error and pass an empty Triple in case of error (I do not know the use-case)

Reply:
Now it throw the exception and not need to pass empty object.